NHibernate.Caches.Common/ConfigurationProviderBase.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Configuration;
+using System.Xml;
+
+namespace NHibernate.Caches.Common
+{
+	/// <summary>
+	/// Base generic class for the cache configuration settings.
+	/// </summary>
+	/// <typeparam name="TConfig">The configuration class.</typeparam>
+	/// <typeparam name="TConfigHandler">The configuration class section handler.</typeparam>
+	public abstract class ConfigurationProviderBase<TConfig, TConfigHandler>
+		where TConfig : class
+		where TConfigHandler : ICacheConfigurationSectionHandler, new()
+	{
+		private static ConfigurationProviderBase<TConfig, TConfigHandler> _current;
+		private static readonly string ConfigurationSectionName;
+
+		static ConfigurationProviderBase()
+		{
+			ConfigurationSectionName = new TConfigHandler().ConfigurationSectionName;
+		}
```

Comment:
I dislike the pattern I have ended up with here, but I have not found a better way to avoid re-coding this for each provider.